orchagent/muxorch.cpp

```diff
@@ -795,22 +804,48 @@ bool MuxNbrHandler::enable(bool update_rt)
         IpPrefix pfx = it->first.to_string();
         if (update_rt)
         {
-            if (remove_route(pfx) != SAI_STATUS_SUCCESS)
-            {
-                return false;
-            }
-            updateTunnelRoute(nh_key, false);
+            bulk_route_ctx_list.push_back(MuxRouteBulkContext(pfx, false));
         }
 
         it++;
     }
 
+    if (update_rt)
+    {
+        if (!createBulkRouteEntries(bulk_route_ctx_list))
+        {
+            gRouteBulker.clear();
+            return false;
+        }
+        gRouteBulker.flush();
+        if (!processBulkRouteEntries(bulk_route_ctx_list))
+        {
+            gRouteBulker.clear();
+            return false;
+        }
+
+        it = neighbors_.begin();
+        while (it != neighbors_.end())
+        {
+            NextHopKey nh_key = NextHopKey(it->first, alias_);
+            if (update_rt)
+            {
+                updateTunnelRoute(nh_key, false);
```

Comment:
Why don't we move this to original section and do this loop only if Bulk api fails. So we don't have to do it everytime

Reply:
I see what you're saying, I think I just wanted to make sure the order was maintained. I'll fix that

Reply:
I think it's safe to take out this loop entirely then, if bulk api fails we will fall back, which will remove the tunnel routes anyways